source/Calamari.Shared/Integration/Packages/Download/MavenPackageDownloader.cs

```diff
@@ -205,7 +204,7 @@ PackagePhysicalFileMetadata SourceFromCache(string packageId, IVersion version,
             Guard.NotNull(mavenPackageId, "mavenPackageId can not be null");
             Guard.NotNull(feedUri, "feedUri can not be null");
 
-            var errors = new HashSet<string>();
+            var errors = new ConcurrentDictionary<string, string>();
```

Comment:
Can you use `ConcurrentBag` here instead?

Reply:
Yeah I went with a dictionary to avoid `Distinct()`, but I don't really mind; have switched it over :+1: